UnitTestBot/UTBotJava
# Utbot-Python refactoring

## Description

Refactored utbot-python test generation structure
* Supported named arguments and named-only arguments
* Supported named arguments in magic-methods
* Fixed code generation for pytest
* Rewrote requirements installation process
* Updated readme
* Removed unused code
* Fixed deep equals asserts code generation
* Updated example testset and created script to generate tests and check coverage (`utbot-python/samples/run_tests.py`)
* Added CLI parameter `--do-not-generate-regression-suite`
* Added CLI parameter `--runtime-exception-behaviour [PASS|FAIL]`

## How to test

### Manual tests

* For named arguments: see `samples/named_arguments/named_arguments.py`: tests should use named arguments (include `__pow__` and `__round__` methods)
* Asserts for `pytest` with `True` / `False` / `None` use `is` instead of `==`
* Before requirements installation utbot should show list of modules and then show notification about success installation

## Self-check list

Check off the item if the statement is true. Hint: [x] is a marked item.

Please do not delete the list or its items.
